app/javascript/spec/retirement-form/retirement-form.spec.js

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import toJson from 'enzyme-to-json';
+import fetchMock from 'fetch-mock';
+import { shallow } from 'enzyme';
+
+import { act } from 'react-dom/test-utils';
+import { mount } from '../helpers/mountForm';
+import RetirementForm from '../../components/retirement-form/index';
+
+const retirementForm = require('../../components/retirement-form/index');
+describe('Retirement Form Component', () => {
+  const retire = {
+    retires_on: new Date('2024-06-08T15:49:04.816Z'),
+    retirement_warn: '',
+    formMode: 'date',
+
+  };
+
+  afterEach(() => {
+    fetchMock.reset();
+    fetchMock.restore();
+  });
+
+  it('should render a new Retirement form', async(done) => {
+    const wrapper = shallow(<RetirementForm retirementID={'["180"]'}/>);
```

Comment:
You should probably wrap this around with an `act`, because the snapshot is not rendering at all...